### PR TITLE
fix(cognito): invalid default for verification email

### LINKED ISSUE #7597: Cognito User Pool: Invalid default for verification email
The default template for a email verification for Cognito User Pools is not valid. It was changed in, I think, 1.26.0, from

    Your verification code is {####}. 

to

    Hello {username}, Your verification code is {####}

However interpolation of `{username}` is not supported in verification emails: https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-message-templates.html

So the message is sent to the user containing `{username}`

Prior this change I think the template was just the default Cognito provided one, but in this change the CDK is setting an explicit template.

### Reproduction Steps

```
    const cognitoUserPool = new UserPool(this, 'CognitoUserPool', {
      userPoolName: 'PoolName',
    });

```

### Error Log

The end user receives a template with {username} in it.

### Environment

  - **CLI Version      : 1.32.0
  - **Framework Version: 1.32.0
  - **OS               : Mac or Linux
  - **Language         : Typescript

### Other

I think this was introduced in this commit:

https://github.com/aws/aws-cdk/commit/faf6693e2590fbe7332be8737afd35874f6719f1

This is the line that generates the default config: https://github.com/aws/aws-cdk/blob/feadd6cb643b415ae002191ba2cb4622221a5af6/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L702
--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by winky**
## Commit Message
fix(cognito): invalid default for verification email (#7790)

The default template contain a placeholder that can not be used in verification email, so remove the place holder

Fixes #7597

BREAKING CHANGE: An invalid template placeholder has been removed
from the default verification email body in a user pool.
## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cognito/lib/user-pool.ts [hunk 1]
```diff
@@ -243,7 +243,7 @@ export interface UserVerificationConfig {
    * See https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-message-templates.html to
    * learn more about message templates.
    *
-   * @default - 'Hello {username}, Your verification code is {####}' if VerificationEmailStyle.CODE is chosen,
+   * @default - 'The verification code to your new account is {####}' if VerificationEmailStyle.CODE is chosen,
    * 'Hello {username}, Verify your account by clicking on {##Verify Email##}' if VerificationEmailStyle.LINK is chosen.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool.ts:247]:
```suggestion
   * 'Verify your account by clicking on {##Verify Email##}' if VerificationEmailStyle.LINK is chosen.
```

File: packages/@aws-cdk/aws-appsync/test/integ.graphql.expected.json [hunk 2]
```diff
@@ -58,7 +58,7 @@
         "UserPoolName": "myPool",
         "VerificationMessageTemplate": {
           "DefaultEmailOption": "CONFIRM_WITH_CODE",
-          "EmailMessage": "Hello {username}, Your verification code is {####}",
+          "EmailMessage": "The verification code to your new account is {####}",
```
Comment (nija-at) [inline packages/@aws-cdk/aws-appsync/test/integ.graphql.expected.json:61]: I believe this message is also used for MFA verification, so perhaps drop the word 'new'. What do you think?

Comment (winky) [reply packages/@aws-cdk/aws-appsync/test/integ.graphql.expected.json:61]: I think your suggestion is great.

Comment (winky) [reply packages/@aws-cdk/aws-appsync/test/integ.graphql.expected.json:61]: @nija-at I think your suggestion is great, but existing code about `EmailSubject` inculde 'new'. Should I drop the word 'new' from `EmailSubject` too?

https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-cognito/lib/user-pool.ts#L698

Comment (nija-at) [reply packages/@aws-cdk/aws-appsync/test/integ.graphql.expected.json:61]: Actually, I am wrong here. Email is not used as MFA; I confused this one with SMS. I'm ok leaving this as it is.

**Pull request merged**